airbyte-cdk/python/airbyte_cdk/sources/connector_state_manager.py

```diff
@@ -47,9 +49,14 @@ def get_stream_state(self, stream_name: str, namespace: Optional[str]) -> Mutabl
         :param namespace: Namespace of the stream being fetched
         :return: The per-stream state for a stream
         """
-        stream_state = self.per_stream_states.get(HashableStreamDescriptor(name=stream_name, namespace=namespace))
+        stream_state: AirbyteStateBlob = self.per_stream_states.get(HashableStreamDescriptor(name=stream_name, namespace=namespace))
         if stream_state:
-            return stream_state.dict()  # type: ignore # mypy thinks dict() returns any, but it returns a dict
+            # TODO: check construction; stream_state should be json here; flaky tests??
```

Comment:
did you figure this out?

Reply:
flaky test_builder;
fixed in https://github.com/airbytehq/airbyte/commit/b1e89a97f4f3bd6b3267ea48d09f878600776d05